unimport/refactor.py

```diff
@@ -1,137 +1,111 @@
-        for imp in self.unused_modules:
+    def get_imp(self, import_name, location):
+        for imp in self.imports:
             if (
                 imp["name"] == import_name
-                and imp["lineno"] == node.get_lineno()
+                and imp["lineno"] == location.start.line
             ):
                 return imp
 
+    def get_location(self, node):
+        return self.get_metadata(PositionProvider, node)
+
+    def leave_import_alike(self, original_node, updated_node):
+        location = self.get_location(original_node)
+        if isinstance(updated_node.names, cst.ImportStar):
+            if isinstance(updated_node.module, cst.Attribute):
+                import_name = self.get_import_name_from_attr(
+                    attr_node=updated_node.module
+                )
+            else:
+                import_name = updated_node.module.value
+            imp = self.get_imp(import_name=import_name, location=location)
+            if imp["modules"]:
+                modules = ",".join(imp["modules"])
+                names_to_suggestion = []
+                for module in modules.split(","):
+                    names_to_suggestion.append(
+                        cst.ImportAlias(cst.Name(module))
+                    )
+                return updated_node.with_changes(names=names_to_suggestion)
+            else:
+                if imp["module"]:
+                    return cst.RemoveFromParent()
+        else:
+            names_to_keep = []
+            for import_alias in updated_node.names:
+                if isinstance(import_alias.name, cst.Attribute):
+                    import_name = self.get_import_name_from_attr(
+                        attr_node=import_alias.name
+                    )
+                else:
+                    import_name = (
+                        import_alias.asname or import_alias
+                    ).name.value
+                if self.is_import_used(import_name, location):
+                    names_to_keep.append(
+                        import_alias.with_changes(
+                            comma=cst.MaybeSentinel.DEFAULT
+                        )
+                    )
+            if len(names_to_keep) == 0:
+                return cst.RemoveFromParent()
+            else:
+                return updated_node.with_changes(names=names_to_keep)
+        return updated_node
+
+    def leave_Import(
+        self, original_node: cst.Import, updated_node: cst.Import
+    ) -> cst.Import:
+        return self.leave_import_alike(original_node, updated_node)
 
-class RefactorTool(RefactoringTool):
-    def __init__(self):
-        self._fixer = RefactorImports()
-        self._fixers = [self._fixer]
-        super().__init__(None, options={"print_function": True})
+    def leave_ImportFrom(
+        self, original_node: cst.ImportFrom, updated_node: cst.ImportFrom
+    ) -> cst.ImportFrom:
+        return self.leave_import_alike(original_node, updated_node)
```

Comment:
or better, just use this functions instead of re-directing everything to a single place. If there is common code between them just put it to a singular function.